arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -824,8 +957,12 @@ public sealed class Option<out A> {
    *   none.exists { it > 10 }      // Result: null
    * }
    * ```
-   * <!--- KNIT example-option-22.kt -->
+   * <!--- KNIT example-option-25.kt -->
    */
+  @Deprecated(
+    NicheAPI + "Prefer Kotlin nullable syntax instead",
+    ReplaceWith("getOrNull()?.takeIf(predicate)")
+  )
   public inline fun findOrNull(predicate: (A) -> Boolean): A? {
```

Comment:
Pretty crazy.. these APIs predate the introduction of `takeIf` into Kotlin Std. Arrow is getting old, happy it's getting a new lack of paint 😍